staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -371,6 +411,71 @@ func (o *DebugOptions) generateDebugContainer(pod *corev1.Pod) *corev1.Ephemeral
 	return ec
 }
 
+// generateNodeDebugPod generates a debugging pod that schedules on the specified node.
+// The generated pod will run in the host PID, Network & IPC namespaces, and it will have the node's filesystem mounted at /node.
+func (o *DebugOptions) generateNodeDebugPod(node string) *corev1.Pod {
+	cn := "debugger"
+	// Setting a user-specified container name doesn't make much difference when there's only one container,
+	// but the argument exists for pod debugging so it might be confusing if it didn't work here.
+	if len(o.Container) > 0 {
+		cn = o.Container
+	}
+
+	// The name of the debugging pod is based on the target node, and it's not configurable to
+	// limit the number of command line flags. There may be a collision on the name, but this
+	// should be rare enough that it's not worth the API round trip to check.
+	pn := fmt.Sprintf("node-debugger-%s-%s", node, nameSuffixFunc(5))
+	if !o.Quiet {
+		fmt.Fprintf(o.ErrOut, "Creating debugging pod %s with container %s on node %s.\n", pn, cn, node)
+	}
+
+	p := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pn,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:                     cn,
+					Env:                      o.Env,
+					Image:                    o.Image,
+					ImagePullPolicy:          o.PullPolicy,
+					Stdin:                    o.Interactive,
+					TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+					TTY:                      o.TTY,
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/node",
```

Comment:
Nit: aybe just `/host` here, and `host`  below

Reply:
SGTM. I've updated this and the other messages to use "node" only in the context of the API object and "host" in the context of namespaces and filesystem. Now that you point it out, I realize this is more in line with how these terms are used in the rest of the API.